### ISSUE TITLE
Add second folder

### ISSUE THREAD

**Issue by Tekbr** [Microsoft/vscode#30608]
- VSCode Version: 1.15.0 Insiders (56a5bf1, Date 2017-07-13T06:29:54.260Z)
- OS Version: Windows 10

-- Sorry for the English, I used Google Translator --

Sorry for the question but ..

**Question 1**- But why am I forced to create a work area?

I simply want to add a second folder, without having to create a specific workspace.

Also when I create the work area it changes the color of the Status bar. (I do not think that would ask it to happen).

This window should have a third option, "Add only the folder".

![workspace](https://user-images.githubusercontent.com/19866231/28165944-632cd3c2-67ac-11e7-9ea2-4f9ce97d545d.PNG)

That is, the options would be: "New Workspace", "Add only the folder", "Cancel".

**Question 2** - What's more, why the "Close Folder" option changed "Close Workspace", even though a Workspace has not yet been created? The menu should change only when a workspace was created, right?

![menunew](https://user-images.githubusercontent.com/19866231/28165953-6aa98424-67ac-11e7-83ea-bdfcf374039d.PNG)

If you need to create a separate issue, please move one of them.

Reproduces without extensions: Yes

**Comment by bpasero** [comment]
@Tekbr thanks for your feedback, this UI is very young and might change until the final bits are out. We want to make workspaces an explicit concept for multi root. You cannot have multiple folders in the classical single-folder views currently. Creating a workspace will bring you into a new experience (hence the status bar change).

I know this is different from how it works in stable currently, but we had lengthy design discussions around the flaws of the model in stable and decided to go with this approach.

**Comment by bpasero** [comment]
@Tekbr see https://github.com/Microsoft/vscode/issues/396#issuecomment-315079618

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

@bpasero Thanks for the quick response, for the attention and patience. Yes I know it's at the beginning, but I found it a bit strange. But I'll wait for the news (said by you in the [#396 (comment)](https://github.com/Microsoft/vscode/issues/396#issuecomment-315079618))

-- --
-- My thought, perhaps the community has another vision --

I think a Workspace should be a project that has several folders. But sometimes we are upgrading the old project to a new project (or querying a similar project folder), and we want to add the old project folder, but without the need to bind.
-- --


Another note (which will probably change in the future): Unsaved workspace should not be placed in Recent Open until it is saved. By clicking "Close Workspace" and if it is not saved, it would be interesting to ask if you would like to save. If the user clicks No, it should not be placed in the menu.

Why that? Because the menu will be very polluted. In addition, you can not differentiate between "Untitled Workspace".

**Comment by bpasero** [comment]
Yeah we think about asking you to save a untitled workspace once you close the window. Good point about the fact that untitled workspaces should not show up in the recent history 👍

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

> Yeah we think about asking you to save a untitled workspace once you close the window. 

@bpasero I think the question should only remain when closing the Workspace, not when closing the window.

If I open another folder with VSCode (right button> Open With VSCode) - it can not be running - and if in the last session it was open an Untitled Workspce, maybe an alert asking if: I want to save the last Workspace before opening The new folder or discard the old Workspace and open the new folder or open last session before (and discard the current folder.)

**Comment by bpasero** [comment]
@Tekbr our current thinking is to ask as soon as you close the window, otherwise we would need to introduce yet another way of dismissing a "Untitled Workspace".

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

@bpasero I think it is best to discard only when we click on "Close Workspace", or the situation described above.

Recalling that within a Workspace there may be unsaved file and Untitled file. In this case it would be good to alert that modified and / or unsaved files (Untitled) could be lost. Or is there no such possibility? Once a workspace is saved, does it matter all the data (including files are saved)?

Or am I saying something silly? Sorry for anything...

**Comment by bpasero** [comment]
@Tekbr sure, dirty files would also cause us to ask what to do (save), I would not change that behaviour. 

If we decide to not cleanup an untitled workspace when you are not closing the window, we would still need to show all untitled workspaces in the "File > Open Recent" menu, how do you otherwise get back to them to close them? And we would need even more UI to dismiss an untitled workspace without opening it to manage them. I find that a lot of UI which can be avoided by just asking you to save the workspace when you close the window?

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

@bpasero  _By what I understand (and according to Google Translate) ..._


> If we decide to not cleanup an untitled workspace when you are not closing the window, we would still need to show all untitled workspaces in the "File > Open Recent" menu, how do you otherwise get back to them to close them?

(I do not know if it's possible) - thought

Currently if I close the VSCode with a folder, and when I open it again it returns no was it, correct?

If you close the VSCode with an untitled workspace when opening the VSCode again it should open as closed, that is, with the untitled workspace.

If it detects that a new folder has been opened (other than the workspace) it should say 'I want to save the last workspace before opening the new folder' or if 'I want to discard the workspace and open the new folder' or 'go back to the last workspace And discard the current folder '. (As I said earlier)


Or option would be:

Limit to 2 or 3 untitled workspace, when the next one is created (the 4 untittled) would ask what you want to save or warn that the oldest will be deleted and you want to continue.

Showing next to the name untitled the date / time it was created (DD / MM / YYYY - HH: MM: SS) in File recent



> I find that a lot of UI which can be avoided by just asking you to save the workspace when you close the window?

In this case you are obliged to save something that is not definitive that has no bond, but that I will use again in some mometo of the day or for some days.


Another idea (even to get more organized) would be to create "File> Open Recent Workspace"

**Comment by bpasero** [comment]
@Tekbr it depends on the OS: On Windows and Linux if you close the last window, we close VS Code and in that case we would not ask to save the workspace and simply restore it the next time you open. However, on macOS, closing the last Window does not quit the application so we would ask to save the workspace.

This feature should work in the same way as "hot exit" works for files today imho.

Good idea about showing the date for untitled workspaces. However, I am not even convinced yet that untitled workspaces should show up under the "Files > Open Recent" category...

**Comment by bpasero** [comment]
All these changes landed, I suggest to retest this and give it individual issues when seeing something. The smooth transition from 1 to 2 folders is something we have on our backlog.

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

@bpasero Sorry for the delay...

> However, on macOS, closing the last Window does not quit the application so we would ask to save the workspace.

Well ... I've never used MacOS so far.
But from what I understand (according to Google Translate), if I want to close the VSCode window, in MacOS does it keep it open, closing like it's a folder in Windows or Linux? Is that it? Strange ...

So I suggested reopening works on Windows and Linux, but it does not work on MacOS. Would that be it?

It may be complex, but why not do it for Windows and Linux, since it works well for both? Some things have already been done where it only works for Mac or better on Mac.

> Good idea about showing the date for untitled workspaces.

As for the date / time do not forget to use the format according to OS settings.

> All these changes landed, I suggest to retest this and give it individual issues when seeing something. The smooth transition from 1 to 2 folders is something we have on our backlog.

Ok I'll check out the latest version of the Insiders
